eo-parser/src/main/java/org/eolang/parser/EoParserErrors.java

```diff
@@ -73,66 +67,76 @@ public void syntaxError(
-            final Parser parser = (Parser) recognizer;
-            final String rule = parser.getRuleInvocationStack().get(0);
-            final String[] names = parser.getRuleNames();
-            final String detailed;
-            if (names[EoParser.RULE_objects].equals(rule)) {
-                detailed = "Invalid object list declaration";
-            } else if (names[EoParser.RULE_metas].equals(rule)) {
-                detailed = "Invalid meta declaration";
-            } else if (names[EoParser.RULE_program].equals(rule)) {
-                detailed = "Invalid program declaration";
-            } else if (names[EoParser.RULE_bound].equals(rule)) {
-                detailed = "Invalid bound object declaration";
-            } else if (names[EoParser.RULE_object].equals(rule)) {
-                detailed = "Invalid object declaration";
-            } else {
-                detailed = "no viable alternative at input";
-            }
-            msgs.add(new MsgLocated(line, position, detailed).formatted());
-            msgs.add(
-                new MsgUnderlined(
-                    this.lines.line(line),
-                    position,
-                    Math.max(token.getStopIndex() - token.getStartIndex(), 1)
-                ).formatted()
-            );
+            result = getDetailedNoViableAlt((Parser) recognizer);
+            underlined = this.getMsgUnderlined(symbol, line, position);
         } else if (error instanceof InputMismatchException) {
-            final Token token = (Token) symbol;
-            final Parser parser = (Parser) recognizer;
-            final String rule = parser.getRuleInvocationStack().get(0);
-            final String detailed;
-            final String[] names = parser.getRuleNames();
-            if (names[EoParser.RULE_program].equals(rule)) {
-                detailed =
-                    "We expected the program to end here but encountered something unexpected";
-            } else if (names[EoParser.RULE_objects].equals(rule)) {
-                detailed =
-                    "We expected a list of objects here but encountered something unexpected";
-            } else {
-                detailed = msg;
-            }
-            msgs.add(new MsgLocated(line, position, detailed).formatted());
-            msgs.add(
-                new MsgUnderlined(
-                    this.lines.line(line),
-                    position,
-                    Math.max(token.getStopIndex() - token.getStartIndex(), 1)
-                ).formatted()
-            );
+            result = getDetailedInputMismatch((Parser) recognizer, msg);
+            underlined = this.getMsgUnderlined(symbol, line, position);
         } else {
-            msgs.add(new MsgLocated(line, position, msg).formatted());
-            msgs.add(this.lines.line(line));
+            result = msg;
+            underlined = this.lines.line(line);
         }
+        final List<String> msgs = List.of(
```

Comment:
@vl4ds4m It seems, we don't need this variable. Can we inline this directly to the following statement?

```java
this.errors.add(new ParsingException(error, line, msgs));
```

https://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html